### PR TITLE
Configured trace config to read from env vars/sys props

### LINKED ISSUE #374: Add support to configure trace sampling probability
Right now all traces are exported by default. Adding a configuration property `ota.trace.sampling.probability` will allow configuring the sampling probability to support basic sampling use cases. 

`ota.trace.sampling.probability` can have below values

- `1 (Default)` - ALWAYS_ON sampler will be used which is the current default behavior
- `0` - ALWAYS_OFF sampler will be used
- `> 0 && < 1` - Probability sampler will be used
- `< 0 || > 1` - invalid value

### PR DESCRIPTION
**Pull request by RashmiRam**
Added `SpringBootWithSamplingSmokeTest.groovy`in `springboot` smoke tests to test sampling. 
@iNikem @trask Do you see any other way to test this configured sampling? 

Closes #374

### CODE REVIEW

File: smoke-tests/springboot/src/test/groovy/io/opentelemetry/smoketest/SpringBootWithSamplingSmokeTest.groovy [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    processBuilder.environment().put("OTEL_CONFIG_SAMPLER_PROBABILITY", "${SAMPLER_PROBABILITY}")
+    processBuilder.directory(new File(buildDirectory))
+  }
+
+  def "default home page with probability sampling enabled"() {
+    setup:
+    def spanCounter = new SpanCounter(logfile, [
+      (HANDLER_SPAN): SPAN_COUNT_TARGET,
+      (SERVLET_SPAN): SPAN_COUNT_TARGET,
+    ], 10000)
```
[the remark this answers is not in the document]
Comment (RashmiRam) [reply smoke-tests/springboot/src/test/groovy/io/opentelemetry/smoketest/SpringBootWithSamplingSmokeTest.groovy]: Yes. This makes sense. Done.

File: smoke-tests/springboot/src/test/groovy/io/opentelemetry/smoketest/SpringBootWithSamplingSmokeTest.groovy [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import okhttp3.Request
+
+class SpringBootWithSamplingSmokeTest extends AbstractServerSmokeTest {
+
+  static final HANDLER_SPAN = "LOGGED_SPAN WebController.greeting"
+  static final SERVLET_SPAN = "LOGGED_SPAN /greeting"
+  static final double SAMPLER_PROBABILITY = 0.2
+  static final int NUM_TRIES = 1000
+  static final int ALLOWED_DEVIATION = 0.1 * NUM_TRIES
+  static final int SPAN_COUNT_TARGET = (SAMPLER_PROBABILITY + 0.1) * NUM_TRIES
```
Comment (iNikem) [inline smoke-tests/springboot/src/test/groovy/io/opentelemetry/smoketest/SpringBootWithSamplingSmokeTest.groovy:27]: This is now unused?

File: agent-tooling/src/main/java/io/opentelemetry/auto/tooling/TracerInstaller.java [hunk 3]
```diff
@@ -75,6 +76,14 @@ private static synchronized void installExportersFromJar(final String exporterJa
               .readEnvironmentVariables()
               .readSystemProperties()
               .build();
+      TraceConfig activeTraceConfig = OpenTelemetrySdk.getTracerProvider().getActiveTraceConfig();
```
Comment (iNikem) [inline agent-tooling/src/main/java/io/opentelemetry/auto/tooling/TracerInstaller.java:79]: I am not sure this is the right place for this code. This method is called `installExportersFromJar`, does not seem to be related to sampling. I would create a new method, e.g. `configure` or `configureSampler`, and call it from `installAgentTracer` method. Independently of exporter.

Comment (trask) [reply agent-tooling/src/main/java/io/opentelemetry/auto/tooling/TracerInstaller.java:79]: good thought, makes sense 👍

Comment (RashmiRam) [reply agent-tooling/src/main/java/io/opentelemetry/auto/tooling/TracerInstaller.java]: Done

Comment (iNikem) [reply agent-tooling/src/main/java/io/opentelemetry/auto/tooling/TracerInstaller.java]: Currently `configure` method is called from `installExportersFromJar`, which is only called when `exporterJar` is configured. Does not seem right to me. Call `configure` from `installAgentTracer`?

Comment (RashmiRam) [reply agent-tooling/src/main/java/io/opentelemetry/auto/tooling/TracerInstaller.java]: Sorry. Totally missed it. Running whole tests suite in local takes more time. I am running `googleJavaFormat`, `integration:test` and `integration:latestDepTest`. Is there any selected set of tests that can be run before pushing?

Comment (iNikem) [reply agent-tooling/src/main/java/io/opentelemetry/auto/tooling/TracerInstaller.java]: Not really :( I don't think we have a set of "important" tests to run when changing such base functionality. Any ideas are very welcomed!

Comment (RashmiRam) [reply agent-tooling/src/main/java/io/opentelemetry/auto/tooling/TracerInstaller.java]: Ah Yes. Agreed that changes in base functionality requires the full suite to be run and I didn't know what exactly needed to be run before raising a PR. I was more looking for a kind of checklist to be verified before raising a PR. Do you think that is useful or Is it just me?

Comment (iNikem) [reply agent-tooling/src/main/java/io/opentelemetry/auto/tooling/TracerInstaller.java]: I think it would be mildly useful :) E.g. I don't ever run `latestDepTest` locally before making PR. `googleJavaFormat` is certainly needed and then just test your changes with corresponding `test` task. Not sure if there are any more required steps.
